yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/credentials/ThirdPartyAttributeConverterTest.java

```diff
@@ -0,0 +1,89 @@
+import com.yoti.api.client.credentials.CredentialIssuanceDetails;
+import com.yoti.api.client.spi.remote.Base64;
+import com.yoti.api.client.spi.remote.proto.IssuingAttributesProto;
+import com.yoti.api.client.spi.remote.proto.ThirdPartyAttributeProto;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.text.ParseException;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ThirdPartyAttributeConverterTest {
+
+    private static final String ENCODED_THIRD_PARTY_ATTRIBUTE = "ChFzb21lSXNzdWFuY2VUb2tlbhIvChgyMDE5LTEwLTE1VDIyOjA0OjA1LjEyM1oSEwoRY29tLnRoaXJkcGFydHkuaWQ=";
+    private static final byte[] ENCODED_THIRD_PARTY_ATTRIBUTE_BYTES = Base64.getDecoder().decode(ENCODED_THIRD_PARTY_ATTRIBUTE);
+
+    private static final IssuingAttributesProto.Definition DEFINITION_PROTO = IssuingAttributesProto.Definition.newBuilder()
+            .setName("com.thirdparty.id")
+            .build();
+
+    private static final IssuingAttributesProto.IssuingAttributes ISSUING_ATTRIBUTES_PROTO = IssuingAttributesProto.IssuingAttributes.newBuilder()
+            .addDefinitions(DEFINITION_PROTO)
+            .setExpiryDate("2019-10-15T22:04:05.123Z")
+            .build();
+
+    private static final ThirdPartyAttributeProto.ThirdPartyAttribute THIRD_PARTY_ATTRIBUTE_PROTO = ThirdPartyAttributeProto.ThirdPartyAttribute.newBuilder()
+            .setIssuingAttributes(ISSUING_ATTRIBUTES_PROTO)
+            .setIssuanceToken(ByteString.copyFromUtf8("someIssuanceToken"))
+            .build();
+
+    private static final byte[] THIRD_PARTY_ATTRIBUTE_BYTES = THIRD_PARTY_ATTRIBUTE_PROTO.toByteArray();
+
+    @Test
+    public void shouldParseValidThirdPartyAttribute() throws Exception {
+        ThirdPartyAttributeConverter thirdPartyAttributeConverter = ThirdPartyAttributeConverter.newInstance();
+        CredentialIssuanceDetails credentialIssuanceDetails = thirdPartyAttributeConverter.parseThirdPartyAttribute(ENCODED_THIRD_PARTY_ATTRIBUTE_BYTES);
+
+        DateTime dateTimeValue = credentialIssuanceDetails.getExpiryDate();
+        String issuanceToken = credentialIssuanceDetails.getToken();
+        List<String> issuingAttributes = credentialIssuanceDetails.getIssuingAttributes();
+
+        assertThat(issuingAttributes, hasSize(1));
+        assertThat(issuanceToken, is("someIssuanceToken"));
+
+        assertEquals(2019, dateTimeValue.getDate().getYear());
+        assertEquals(10, dateTimeValue.getDate().getMonth());
+        assertEquals(15, dateTimeValue.getDate().getDay());
+
+        assertEquals(22, dateTimeValue.getTime().getHour());
+        assertEquals(04, dateTimeValue.getTime().getMinute());
+        assertEquals(05, dateTimeValue.getTime().getSecond());
```

Comment:
If there isn't already a method to check that the datetime matches, do you think it would be worth adding one? So we don't have to use 6 lines to check

Reply:
When I looked through the code, I couldn't find a function that did this.  We could probably create one though that has a method signature something around the lines of:

`assertDateTime(DateValue check, int year, int month, int day, int hour, int minute, int second)`.

Thoughts?

Reply:
Fixed in 5a78f1f

Reply:
👌